### PR TITLE
Add log sharing message on login.

### PR DESCRIPTION
**Pull request by dchw**
~~Hold until after our next release.~~

### CODE REVIEW

**Review by alexcb — APPROVED** [review]

**Review by brandonSc — APPROVED** [review]

File: cmd/earthly/main.go [hunk 1]
```diff
@@ -2492,21 +2495,28 @@ func (app *earthlyApp) actionAccountLogin(c *cli.Context) error {
...
+		app.printLogSharingMessage()
 	}
 	if err = cc.Authenticate(); err != nil {
 		return errors.Wrap(err, "authentication with cloud server failed")
 	}
 	return nil
 }
 
+func (app *earthlyApp) printLogSharingMessage() {
+	app.console.Printf("Log sharing is enabled by default. If you would like to disable it, run:")
```
Comment (alexcb) [inline cmd/earthly/main.go:2516]: nit: move `\n` from below printf up to this line.

Comment (dchw) [reply cmd/earthly/main.go]: I smooshed all the printing together with the intent on making the resulting layout/formatting more explicit.

**Review by alexcb — APPROVED** [review]

**Comment by dchw** [comment]
@alexcb @brandonSc I flipped a few more switches that made sense with this PR.

1. Log sharing is default now
2. Removed un-disabling sharing in GHA actions
3. Fixed non-logged in users message (bad closure + `errors.Is`)

**Review by brandonSc — APPROVED** [review]
🎉

**Pull request merged**